### PR TITLE
Add v3 client library

### PR DESCRIPTION
**Pull request by florindragos**
Added new v3 project
Moved some of the CloudFoundryClient code to common
Added the existent v3 endpoints to the v3 client

### CODE REVIEW

**Comment by viovanov** [comment]
Please add the test project(s) to their respective build.proj tasks

File: src/CloudFoundry.CloudController.Common/CloudFoundryClient.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
+namespace CloudFoundry.CloudController.Common
+{
+    using System;
+    using System.Globalization;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using CloudFoundry.CloudController.Common.Http;
+
+    /// <summary>
+    /// This is the Cloud Foundry client. To use it, you need a Cloud Foundry endpoint.
```
Comment (viovanov) [inline src/CloudFoundry.CloudController.Common/CloudFoundryClient.cs]: This summary is no longer accurate.

File: src/CloudFoundry.CloudController.Common/CloudFoundryClient.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        /// Proxy
+        /// </summary>
+        public Uri HttpProxy { get; protected set; }
+
+        /// <summary>
+        /// Skip Validation
+        /// </summary>
+        public bool SkipCertificateValidation { get; protected set; }
+    }
+}
```
Comment (viovanov) [inline src/CloudFoundry.CloudController.Common/CloudFoundryClient.cs]: InitEndpoints should be declared as an abstract method here.

File: src/CloudFoundry.CloudController.V3.Client.Test/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("CloudFoundry.CloudController.V2.Client.Test")]
```
Comment (viovanov) [inline src/CloudFoundry.CloudController.V3.Client.Test/Properties/AssemblyInfo.cs]: Incorrect title

File: src/CloudFoundry.CloudController.V3.Client.Test/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("CloudFoundry.CloudController.V2.Client.Test")]
```
Comment (viovanov) [inline src/CloudFoundry.CloudController.V3.Client.Test/Properties/AssemblyInfo.cs]: incorrect product

**Comment by viovanov** [comment]
Also append to the sample in the readme - show how to init and do an operation with the V3 client.

**Comment by mihaibuzgau** [comment]
:+1:

**Pull request merged**